tests/cmsis_rtos_v1/src/signal.c

```diff
@@ -69,21 +111,91 @@ void test_signal_events_timeout(void)
 	 */
 	evt = osSignalWait(SIGNAL1, TIMEOUT);
 	zassert_equal(evt.status, osEventTimeout, "");
+	osThreadTerminate(id1);
 }
 
 void test_signal_events_signalled(void)
 {
-	osThreadId id1, id2;
+	osThreadId id1, id2, id3;
 	osEvent evt;
+	int signals;
 
 	id1 = osThreadCreate(osThread(Thread_1), osThreadGetId());
 	zassert_true(id1 != NULL, "Thread creation failed");
 
 	id2 = osThreadCreate(osThread(Thread_2), osThreadGetId());
 	zassert_true(id2 != NULL, "Thread creation failed");
 
-	/* wait for a signal */
+	/* wait for multiple signals */
 	evt = osSignalWait(SIGNAL, TIMEOUT);
 	zassert_equal(evt.status, osEventSignal, "");
 	zassert_equal((evt.value.signals & SIGNAL), SIGNAL, "");
+
+	signals = osSignalClear(osThreadGetId(), SIGNAL);
+	zassert_not_equal(signals, 0x80000000, "");
+
+	/* Set any single signal */
+	signals = osSignalSet(osThreadGetId(), SIGNAL1);
+	zassert_not_equal(signals, 0x80000000, "");
+
+	/* wait for any single signal flag */
+	evt = osSignalWait(0, TIMEOUT);
+	zassert_equal(evt.status, osEventSignal, "");
+	zassert_equal(evt.value.signals, SIGNAL1, "");
+
+	/* validate by passing invalid parameters */
+	zassert_equal(osSignalSet(NULL, 0), 0x80000000,
+				  "NULL signal set unexpectedly");
+	zassert_equal(osSignalClear(NULL, 0), 0x80000000,
+				  "NULL signal cleared unexpectedly");
+	/* cannot wait for Flag mask with MSB set */
+	zassert_equal(osSignalWait((int32_t)0x80000000, 0).status, osErrorValue,
+				  "signal wait passed unexpectedly");
+	zassert_equal(osSignalSet(osThreadGetId(), (int32_t)0x80000000),
```

Comment:
use something like 0x80010000 instead of 0x80000000 to cover a wider test case range.

Reply:
Addressed. Thanks.